src/t_stream.c

```diff
@@ -2450,7 +2451,7 @@ void xreadCommand(client *c) {
  * specified as argument of the function. */
 streamNACK *streamCreateNACK(streamConsumer *consumer) {
     streamNACK *nack = zmalloc(sizeof(*nack));
-    nack->delivery_time = mstime();
+    nack->delivery_time = server.mstime;
```

Comment:
maybe we can skip these changes, and let them be handled by #10300.

Reply:
it's not exactly the same, #10300 talks about using a constant time when handling volatile keys while my changes are mostly for performance

generally speaking, there are many places in the code that access the actual time while the cached time will do, perhaps the default behavior of `commandTimeSnapshot` should be to return the cached time, regardless of `server.fixed_time_expire` (in which case it might become redundant and we can rely on `server.in_nested_call` to know when to update the cached time)

@enjoy-binbin

Reply:
for the record: i can drop this change if https://github.com/redis/redis/pull/10300#discussion_r977267373 is applied

Reply:
somehow i missed this comment, now i think we are good to go (#10300 is applied the change)